python/tank/bootstrap/resolver.py

```diff
@@ -131,36 +133,163 @@ def resolve_shotgun_configuration(
         self,
         pipeline_config_name,
         fallback_config_descriptor,
-        sg_connection
+        sg_connection,
+        current_login
```

Comment:
Any worries about backward compatibility if anybody is using this? Could this be optional?